lib/auth/mocku2f/mocku2f.go

```diff
@@ -149,15 +149,19 @@ type signRegisterResult struct {
 }
 
 func (muk *Key) signRegister(appIDHash, clientDataHash []byte) (*signRegisterResult, error) {
-	//nolint:staticcheck // SA1019 requires Go 1.21 in go.mod and non-insignificant changes. TODO: fix.
-	pubKey := elliptic.Marshal(elliptic.P256(), muk.PrivateKey.PublicKey.X, muk.PrivateKey.PublicKey.Y)
+	// Marshal pubKey into the uncompressed "4||X||Y" form.
+	ecdhPubKey, err := muk.PrivateKey.PublicKey.ECDH()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	pubKey := ecdhPubKey.Bytes()
 
 	var dataToSign []byte
```

Comment:
nit:

```suggestion
        cap := 1 + len(appIDHash) + len(clientDataHash) + len(muk.KeyHandle) + len(pubKey)
	dataToSign := make([]byte, 0, cap)
```

Reply:
Good call, I've done it for the block below and dropped some more of the unnecessary `array[:]` syntax.